inception-project/inception
# #1051 - Switch completely to Java-based Spring configuration

**What's in the PR**
- Replace `security-context.xml` with an approach based on Spring's Java-config capabilities

**How to test manually**
* Log in as a valid user and check if it works
* Try logging in with a bad username / password
* Try accessing the remote API and ensure that it is only accessible by users with `ROLE_REMOTE` (enable `remote-api.enabled=true` in `settings.properties`, access API page via `.../swagger-ui.html`, actually try out invoking an API call - this should bring up a HTTP basic auth dialog in the browser - note that the browser caches the credentials, so try in a private window)

**Automatic testing**

**Documentation**
